_posts/2020-04-11-native-mobile-to-flutter.markdown

```diff
@@ -0,0 +1,122 @@
+---
+layout: post
+title: "Looking Back on Our Migration from Native Mobile to Flutter."
+date: 2020-04-11
+categories: [mobile, flutter]
+author: Bobby Manuel 
+---
+
+### Timeline of the project.
+
+- Project spanned Aug 2019 - Dec 2019 
+- Initial release to Android users in February 2020.
+- Release to all users Android + iOS March 2020.
+- As of the time of this article we've been live across all of our users for about a month with no Flutter related issues.
+
+
+### Why did we do it?
+
+We had two working native mobile applications, but lacked feature parity for Android and iOS users.
+
+Android and iOS development took place at different paces, with varying quality and cost.
+
+Maintaining redundancy across two mobile teams required us to maintain a minimum of four developers on mobile rather than as few as two in a cross platform environment.
+
+### We wanted to achieve...
+
+- feature parity on both platforms.
+- a unified mobile team with higher quality peer reviews. 
+- increased velocity and reduced cost per feature. 
+- higher quality software. 
+- scalability and redundancy of our mobile team.
+
+This seemed possible as the cross-platform mobile space has matured over the last year: there were two great options to choose from, _Flutter_ and _React Native_.
+
+### Why did we choose Flutter? 
+
+The biggest challenge we faced was the timeline within which we had to complete the switch: 5 months.  Our native apps had 140,000 lines of code between the two platforms, so we needed to factor in development speed.
+
+_Regardless of our choice our entire team would have to..._
+
+1. learn a new programming language
+2. learn a new UI framework
+3. add sufficient test coverage to confidently launch the application
+4. delivery a highly secure application. 
+
+Although React Native is a clear winner for a javascript-centric development team, our mobile engineers are not javascript savvy.
+
+#### Here was our decision criteria:
+
+- Community size - _+1 React_
+- Maturity - _+1 React_
+- Learning curve. - _+1 Flutter_
+- Complexity. - _+1 Flutter_
+- Development speed. - _+1 Flutter_
+- Tooling and IDE.  - _+1 Flutter_
+- Testability. - _+1 React_ 
+- Shared code - _+1 Flutter_
+- Platform independence - _+1 Flutter_
+
+
+#### Things we liked about React:
+
+- Maturity and community size: React Native has been in production for much longer.
+- Testing utilities are definitely better on React.
+- More third party library support.
+
+#### Things we liked about Flutter:
+
+- Without javascript developers on our mobile team, the learning curve for Dart/Flutter was substantially lower than Javascript/React Native. 
```

Comment:
Would be great to have more detail here to explain why this is true - e.g. "JS is a crazy language with a lot of warts because...".

Reply:
Or, if you want to stay on the positive side, you can say that Dart is a simpler/cleaner language